D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
-      scope: keyword.other.d
-    - match: \b(new|throws)\b
-      scope: keyword.other.class-fns.d
-    - match: \b(package|extern)\b
-      scope: keyword.other.external.d
-    - match: \b(deprecated|unittest|debug)\b
-      scope: keyword.other.debug.d
-    - match: \b(u_char|u_short|u_int|u_long|ushort|uint|u_quad_t|quad_t|qaddr_t|caddr_t|daddr_t|dev_t|fixpt_t|blkcnt_t|blksize_t|gid_t|in_addr_t|in_port_t|ino_t|key_t|mode_t|nlink_t|id_t|pid_t|off_t|segsz_t|swblk_t|uid_t|id_t|clock_t|size_t|ssize_t|time_t|useconds_t|suseconds_t)\b
-      scope: support.type.sys-types.c
-    - match: \b(pthread_attr_t|pthread_cond_t|pthread_condattr_t|pthread_mutex_t|pthread_mutexattr_t|pthread_once_t|pthread_rwlock_t|pthread_rwlockattr_t|pthread_t|pthread_key_t)\b
-      scope: support.type.pthread.c
-    - match: \b(int8_t|int16_t|int32_t|int64_t|uint8_t|uint16_t|uint32_t|uint64_t|int_least8_t|int_least16_t|int_least32_t|int_least64_t|uint_least8_t|uint_least16_t|uint_least32_t|uint_least64_t|int_fast8_t|int_fast16_t|int_fast32_t|int_fast64_t|uint_fast8_t|uint_fast16_t|uint_fast32_t|uint_fast64_t|intptr_t|uintptr_t|intmax_t|intmax_t|uintmax_t|uintmax_t)\b
-      scope: support.type.stdint.c
-  all-types:
-    - include: support-type-built-ins-d
-    - include: support-type-d
-    - include: storage-type-d
-  comments:
-    - match: /\*
-      scope: punctuation.definition.comment.d
+      set: [value-condition-after, value]
+    - match: '(?=function|delegate)'
+      set: [value-after, basic-type2]
+    - match: '(?=\S)'
+      pop: true
+  value-condition-after:
+    - match: ':'
+      scope: keyword.operator.logical.d
+      set: value
+    - include: not-whitespace-illegal
+  value-parens-after:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      set: value-after
+    - include: not-whitespace-illegal
+  value-bracket-after:
+    - match: '\]'
+      scope: punctuation.section.brackets.end.d
+      set: value-after
+    - include: not-whitespace-illegal
+
+  value-list:
+    - match: '(?=\)|}|]|;)'
+      pop: true
+    - match: '(?=\S)'
+      set: [value-list-after, value]
+  value-list-after:
+    - match: '(?=\)|}|]|;)'
+      pop: true
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: value-list
+    - include: not-whitespace-illegal
+
+  value-array-list:
+    - match: '(?=\])'
+      pop: true
+    - match: '(?=\S)'
+      set: [value-array-list-after, value]
+  value-array-list-after:
+    - match: ':'
+      scope: punctuation.separator.mapping.key-value.d
+      set: [value-array-list-after-value, value]
+    - include: value-array-list-after-value
+  value-array-list-after-value:
+    - match: '(?=\])'
+      pop: true
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: value-array-list
+    - include: not-whitespace-illegal
+
+  alias-in:
+    - match: '\b(alias)\s+({{identifier}})\s+(this)(;)'
+      captures:
+        1: keyword.control.alias.d
+        2: variable.d
```

Comment:
There should probably be a second label on this scope.

Reply:
So I'm not sure about this one. The identifier can refer to anything from a member to a method and can be of literally any type.